README.md

```diff
@@ -1 +1,167 @@
+
+Now where did the other two request go? They are inside of the `processResponse1`:
+
+```js
+callOne({...}, processResponse1)
+
+function processResponse1(error, data1) {
+  callTwo(data1, processResponse2)
+}
+
+function processResponse2(error, data2) {
+  callThere(data2, processResponse3)
+}
+
+function processResponse3(error, data1) {
+  ...
+}
+```
+
+This code is considerably easier to read, but that's not all we can do to improve it. We can take it a step further.
+
+Note: `callOne`, `callTwo`, etc. are just pseudo code for making HTTP requests. In real-life example you would substitute them for real functions using libraries like `http`, `request` or `superagent`. For now it's not really important what those functions do as long as they are asynchronous and use callbacks.
+
+## Using Modules to Mitigate Callback Hell
+
+Imagine having to work on a file with 10 named functions as listed in the previous section. You have 5 co-workers working on the same file. Each callback/response is 50 lines of code. That's insane! Of course it's better than having them in the closures (i.e., anonymous function definitions), but we know how to use modules so let's apply them by making each callback a separate module:
+
+```js
+var processResponse1 = require('./response1.js')
+callOne({...}, processResponse1)
+```
+
+This is the `response1.js` file:
+
+```js
+var processResponse2 = require('./response2.js')
+module.exports = function processResponse1(error, data1) {
+  callTwo(data1, processResponse2)
+}
+```
+
+This is the `response2.js` file:
+
+```js
+var processResponse3 = require('./response3.js')
+module.exports = function processResponse2(error, data2) {
+  callThere(data2, processResponse3)
+}
+```
+
+This is the `response3.js` file:
+
+```js
+module.exports = function processResponse3(error, data3) {
+  ...
+}
+```
+
+
+So now your 5 co-workers will love you even more because each piece of logic will be in its own file (you can easily separate files into different folder by providing paths to the `require`). Another added benefit of modular approach is that you and your team can write unit tests for each module. Having unit tests greatly increases robustness of the system and the velocity of development, because developers can make changes without being afraid of unintentionally breaking existing functionality (if they break the tests will catch that!).
+
+## Callback Arguments Convention
+
+Typically callbacks follow `error, result` convention where `error` is an error object (null is there's no errors) and `result` which is what you would expect from a function. Because this is just a convention, libraries and frameworks don't have to follow it. That's why you must read the usage for each library before attempting to use it's methods.
```

Comment:
Open question: Is it worth noting that this is specifically a Node.js convention? Curious students learning Go, for example, will note that the convention there is exactly the opposite.

cc @vicfriedman @jmburges